tests/containers/rootless_podman.pm

```diff
@@ -0,0 +1,64 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test rootless mode on podman.
+# - add a user on the /etc/subuid and /etc/subgid to allow automatically allocation subuid and subgid ranges.
+# - check uids allocated to user (inside the container are mapped on the host)
+# - give read access to the SUSE Customer Center credentials to call zypper from in the container
+#   proper way to add Access Control List is via `setfacl` which is not available so we just do
+#   `chmod` instead!! This grants the current user the required access rights
+# - Test rootless container:
+#   * container is launched with default root user
+#   * container is launched with existing user id
+#   * container is launched with keep-id of the user who run the container
+# - Restore /etc/zypp/credentials.d/ credentials
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use testapi;
+use utils;
+use containers::common;
+use containers::container_images;
+use suse_container_urls 'get_suse_container_urls';
+use version_utils qw(get_os_release);
+
+sub run {
+    my ($self) = @_;
+    my ($image_names, $stable_names) = get_suse_container_urls();
+    my ($running_version, $sp, $host_distri) = get_os_release;
+    my $runtime = "podman";
+
+    install_podman_when_needed($host_distri);
+    allow_selected_insecure_registries(runtime => $runtime);
+    my $user = $testapi::username;
+    assert_script_run "cat /etc/subuid | grep bernhard || echo /etc/subuid has no uid range for $user";
+    assert_script_run "cat /etc/subgid | grep bernhard || echo /etc/subgid has no gid range for $user";
+    assert_script_run "usermod --add-subuids 200000-201000 --add-subgids 200000-201000 $user";
+    assert_script_run "cat /etc/subuid | grep bernhard", fail_message => "subuid range not assigned for $user";
+    assert_script_run "cat /etc/subgid | grep bernhard", fail_message => "subgid range not assigned for $user";
+    # Workaround instead of
+    # "setfacl -m u:$user:r /etc/zypp/credentials.d/*"
+    assert_script_run "chmod -R 777 /etc/zypp/credentials.d/*";
```

Comment:
`chown $user` would be more restrictive

Reply:
Does it make any difference here. it is not realistic approach. it is used only to do the testing and then i restore the correct bits of the files.

Reply:
Marking config files as executable isn't right in any case IMO

Reply:
i changed it to 666.